### PR TITLE
Merging Blazor feature branch into release/3.1

### PR DESCRIPTION
**Pull request by joperezr**
cc: @ericstj @pranavkm @Jozkee @Anipik @safern 

This was mostly a straight forward merge except for the following conflicts/changes I made:

- **[Merge Conflict]** eng\Versions.props: We had done some special branding for blazor feature branch, I kept source as it was on release/3.1 branch as that is what we wanted.
- **[Merge Conflict]** src\packages.builds: We had conflicts here since the packages that were being built were different. I kept source as it was on release/3.1.
- **[Post-Merge manual change]** eng\corefx-base.yml and eng\windows.yml: We made some changes on the blazor feature branch in order to ONLY ship the http json package on these yml files. I removed those changes and kept source as it was on release/3.1
- **[Post-Merge manual change]** packageIndex.json: I manually added the 3.2.0 as stable version, as it will have shipped by the time we merge this PR

Other than the above details, this was just a direct merge and I didn't have to edit anything else.

### CODE REVIEW

**Comment by joperezr** [comment]
Adding No-Merge label to block merging until branch is open for changes.

**Review by ericstj — APPROVED** [review]
LGTM

We'll want another PR to enable build of System.Net.Http.Json to ship an update.

File: src/System.Net.Http.Json/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿<Project>
+  <Import Project="..\Directory.Build.props" />
+  <PropertyGroup>
+    <StrongNameKeyId>Open</StrongNameKeyId>
+    <AssemblyVersion>3.2.0.0</AssemblyVersion>
+    <PackageVersion>3.2.0</PackageVersion>
```
Comment (safern) [inline src/System.Net.Http.Json/Directory.Build.props:6]: Should package/assembly version be `3.2.1`?

Comment (ericstj) [reply src/System.Net.Http.Json/Directory.Build.props:6]: We don't do that until we're ready to ship a new version, which is what I was alluding to in the above comment.  We shouldn't mix feature work with a merge PR.

Comment (safern) [reply src/System.Net.Http.Json/Directory.Build.props:6]: I see, makes sense.

**Review by safern — APPROVED** [review]

**Review by Anipik — APPROVED** [review]

**Pull request merged**